roles/openshift_facts/library/openshift_facts.py

```diff
@@ -534,6 +534,27 @@ def set_aggregate_facts(facts):
 
     return facts
 
+def set_openstack_facts(facts):
+    """ Set openstack facts
+
+        Args:
+            facts (dict): existing facts
+        Returns:
+            dict: the facts dict updated with openstack facts
+    """
+
+    def _add_openstack_config(conf):
+        conf.setdefault('cloud-provider', []).append('openstack')
+        conf.setdefault('cloud-config', []).append('/etc/cloud.conf')
+
+    if 'master' in facts:
+        _add_openstack_config(facts['master'].setdefault('api_server_args', {}))
+        _add_openstack_config(facts['master'].setdefault('controller_args', {}))
+
+    if 'node' in facts:
+        _add_openstack_config(facts['node'].setdefault('kubelet_args', {}))
+
+    return facts
```

Comment:
@lebauce I'm thinking something along the lines of this...

In openshift_facts, within one of the methods that deals with openstack provider facts (possibly normalize_openstack_facts), set the following:
```
for arg in ('api_server_args', 'controller_args', 'kubelet_args'):
    facts['provider'][arg] = {'cloud-provider': ['openstack'], 'cloud-config': ['/etc/cloud.conf']}
```
Though looking at that now, we could probably just set it in normalize_provider_facts and do something like this:
```
for arg in ('api_server_args', 'controller_args', 'kubelet_args'):
    facts['provider'][arg] = {'cloud-provider': [provider], 'cloud-config': ['/etc/cloud.conf']}
```

Then there would need to be an additional method added (say merge_cloud_provider_args) that is called from generate_facts after the merge_facts method:
```
def merge_cloud_provider_args(self, facts):
    if 'provider' in facts:
        if 'master' in facts:
            for arg in ('api_server_args', 'controller_args'):
                if arg in facts['master'] and arg in facts['provider']:
                    for k,v in facts['provider'][arg].iteritems():
                        if k not in facts['master'][arg]:
                            facts['master'][arg][k] = v
        if 'node' in facts:
            if 'kubelet_args' in facts['node'] and 'kubelet_args' in facts['provider']:
                for k,v in facts['provider']['kublet_args'].iteritems():
                    if k not in facts['node']['kubelet_args']:
                        facts['node']['kubelet_args'][k] = v
```
That way, the user can still override the values if they chose, but the default values will be set if they do not.